### PR TITLE
feat: add Progress component

### PR DESCRIPTION
**Pull request by HugoPerard**
Adding a new component : **Progress**
This component is composable and split in three component : 

1. Progress: the parent one
2. ProgressLabel: child that permits to display a label, with Progress informations
3. ProgressBar: child display Progress Chakra component

Capture d’écran 2022-02-25 à 15 58 40

### CODE REVIEW

File: src/components/Progress/Progress.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import { Stack, StackProps } from '@chakra-ui/react';
+
+export const ProgressContext = createContext<ProgressValues>(null);
+
+export interface ProgressValues {
+  completed: number;
+  total: number;
+  isLoading?: boolean;
+}
```
Comment (yoannfleurydev) [inline src/components/Progress/Progress.tsx:11]: I'd swap the two blocks to declare the ProgressValues before using them.

```suggestion
export interface ProgressValues {
  completed: number;
  total: number;
  isLoading?: boolean;
}

export const ProgressContext = createContext<ProgressValues>(null);
```

File: src/components/Progress/Progress.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+}) => {
+  return (
+    <ProgressContext.Provider
+      value={{
+        completed,
+        total,
+        isLoading,
+      }}
+    >
+      <Stack spacing={1} {...rest} />
```
Comment (yoannfleurydev) [inline src/components/Progress/Progress.tsx:29]:
```suggestion
      <Stack {...rest} />
```

File: src/components/Progress/Progress.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    </ProgressContext.Provider>
+  );
+};
```
Comment (yoannfleurydev) [inline src/components/Progress/Progress.tsx:32]:
```suggestion
};

export const useProgressContext = () => useContext(ProgressContext);
```

File: src/components/Progress/ProgressBar.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import { FC, useContext } from 'react';
+
+import { Progress, ProgressProps } from '@chakra-ui/react';
+
+import { ProgressContext } from '.';
+
+export const ProgressBar: FC<ProgressProps> = (props) => {
+  const { completed, total, isLoading } = useContext(ProgressContext);
```
Comment (yoannfleurydev) [inline src/components/Progress/ProgressBar.tsx:8]:
```suggestion
  const { completed, total, isLoading } = useProgressContext();
```

File: src/components/Progress/ProgressLabel.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+  children,
+  ...others
+}) => {
+  const progressData = useContext(ProgressContext);
+  return (
+    <Text fontWeight="medium" {...others}>
+      {children(progressData)}
+    </Text>
+  );
+};
```
Comment (yoannfleurydev) [inline src/components/Progress/ProgressLabel.tsx:21]:
```suggestion
export const ProgressLabel: FC<ProgressLabelProps> = ({
  children,
  ...rest
}) => {
  const progressValues = useProgressContext();

  return (
    <Text fontWeight="medium" {...rest}>
      {children(progressValues)}
    </Text>
  );
};
```

**Review by yoannfleurydev — CHANGES_REQUESTED** [review]

**Comment by ivan-dalmet** [comment]
Thanks for the PR but we will not merge it.
It's more a project specific component because it can be done quickly with same amount of code with Chakra Components & theming :)

```jsx
<Stack spacing="1">
  <Text>10 tasks completed on 30</Text>
  <Progress size="xs" value={10/30*100} />
</Stack>
```

```jsx
<Stack spacing="1">
  <Text>Loading...</Text>
  <Progress size="xs" isIndeterminate />
</Stack>
```

**Pull request closed without merging**